signingmodules/kaleidokms/internal/keymanagersigningmodule/key_manager_signing_module.go

```diff
@@ -0,0 +1,318 @@
+	}
+
+	var keyPath string
+	for _, segment := range path {
+		keyPath = fmt.Sprintf("%s%s/", keyPath, segment.Name)
+	}
+
+	return fmt.Sprintf("kld:///keystore/%s/key/%s%s", keystore, keyPath, keyName), nil
+}
+
+func (rsm *keyManagerSigningModule) ResolveKey(ctx context.Context, req *prototk.ResolveKeyRequest) (*prototk.ResolveKeyResponse, error) {
+	if len(req.Name) == 0 {
+		return nil, i18n.NewError(ctx, msgs.MsgResolveKeyCannotBeEmpty)
+	}
+
+	if len(req.RequiredIdentifiers) == 0 {
+		return nil, i18n.NewError(ctx, msgs.MsgResolveKeyMissingIdentifiers)
+	}
+
+	if (req.RequiredIdentifiers[0].VerifierType != verifiers.ETH_ADDRESS) || (req.RequiredIdentifiers[0].Algorithm != algorithms.ECDSA_SECP256K1) {
+		return nil, i18n.NewError(ctx, msgs.MsgResolveKeyInvalidIdentifierType)
+	}
+
+	uri, err := rsm.GetURI(ctx, req.Name, req.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	log.L(ctx).Debugf("KeyManagerSigningModule ResolveKey uri: %s ", uri)
+
+	keyManagerResolveRequest := ResolveKeyRequest{
+		KeyIdentifier: &KeyIdentifier{
+			KeyURI: &uri,
+		},
+		PublicIdentifierTypesToResolve: []string{
+			ADDRESS_ETH,
+		},
+	}
+
+	// Set autoKeyCreation and KeySpec if autoKeyCreation is enabled
+	if rsm.autoKeyCreation {
+		keyManagerResolveRequest.AutoKeyCreation = true
+		keyManagerResolveRequest.KeySpec = EC_NIST_P256
+	}
+
+	var response ResolveKeyResponse
+	var errInfo ErrorResponse
+
+	res, err := rsm.httpClient.R().
+		SetContext(ctx).
+		SetBody(keyManagerResolveRequest).
+		SetResult(&response).
+		SetError(&errInfo).
+		Post("/resolve")
+
+	if err != nil || (!res.IsSuccess() && res.StatusCode() != http.StatusNotFound) {
+		// Check if this is the KA058002 error (key doesn't have public identifier of type address_ethereum)
+		if rsm.autoKeyCreation && errInfo.Error != "" && strings.Contains(errInfo.Error, "KA058002") {
+			return rsm.handleMissingPublicIdentifier(ctx, uri)
+		}
+		return nil, rsm.kmsRequestError(ctx, err, errInfo)
+	}
+
+	if res.StatusCode() == http.StatusNotFound {
+		return nil, i18n.NewError(ctx, msgs.MsgKMSKeyNotFound, uri)
+	}
+
+	if len(response.Identifiers) == 0 {
+		return nil, i18n.NewError(ctx, msgs.MsgResolveKeyNoIdentifiersResponse, *keyManagerResolveRequest.KeyIdentifier.KeyURI)
+	}
+
+	return &prototk.ResolveKeyResponse{
+		KeyHandle: response.ID,
+		Identifiers: []*prototk.PublicKeyIdentifier{
+			{
+				Algorithm:    algorithms.ECDSA_SECP256K1,
+				VerifierType: verifiers.ETH_ADDRESS,
+				Verifier:     response.Identifiers[0].Value,
+			},
+		}}, nil
+}
+
+// handleMissingPublicIdentifier handles the case where a key exists but doesn't have a public identifier
+// of type address_ethereum. It creates the public identifier and returns the response.
+func (rsm *keyManagerSigningModule) handleMissingPublicIdentifier(ctx context.Context, uri string) (*prototk.ResolveKeyResponse, error) {
+	// First, make a second resolve request to get the key ID
+	keyManagerResolveRequest := ResolveKeyRequest{
```

Comment:
This feels like a very weird interface on the Keystore V2 side.

When I come to other integrations, I'm unlikely to accept this on the client side - and would likely push it to the server side.

Why would our Keystore V2 API want users to need to do `KA058002` detection, and then a 2nd API call?

Surely like `autoKeyCreation` + `EC_NIST_P256`, it's even more clear we should support `autoPublicVerifierCreation` / `eth_address`